functions/webhooks/twitter/FlexToTwitter.protected.ts

```diff
@@ -68,60 +67,57 @@ export const handler = async (
   const resolve = bindResolve(callback)(response);
 
   try {
-    const { recipientId, Body } = event;
-
+    const { recipientId, Body, From, ChannelSid, EventType, Source } = event;
     if (recipientId === undefined) {
       resolve(error400('recipientId'));
       return;
     }
-
     if (Body === undefined) {
       resolve(error400('Body'));
       return;
     }
-
-    console.log('------ FlexToTwitter excecution ------');
-
-    if (event.Source === 'SDK') {
-      const TwitResponse = await sendTwitterMessage(context, recipientId, Body);
-      console.log('Message sent from SDK call: ', Body);
-      resolve(success(TwitResponse));
+    if (From === undefined) {
+      resolve(error400('From'));
+      return;
+    }
+    if (ChannelSid === undefined) {
+      resolve(error400('ChannelSid'));
+      return;
+    }
+    if (EventType === undefined) {
+      resolve(error400('EventType'));
+      return;
+    }
+    if (Source === undefined) {
+      resolve(error400('Source'));
       return;
     }
 
-    if (event.Source === 'API' && event.EventType === 'onMessageSent') {
-      const { ChannelSid } = event;
+    console.log('------ FlexToTwitter excecution ------');
 
-      if (ChannelSid === undefined) {
-        resolve(error400('ChannelSid'));
-        return;
-      }
+    const handlerPath = Runtime.getFunctions()['helpers/customChannels/flexToCustomChannel'].path;
+    const flexToCustomChannel = require(handlerPath) as FlexToCustomChannel;
 
-      const client = context.getTwilioClient();
-      const channel = await client.chat
-        .services(context.CHAT_SERVICE_SID)
-        .channels(ChannelSid)
-        .fetch();
+    const sanitizedEvent = { Body, From, ChannelSid, EventType, Source };
 
-      const channelAttributes = JSON.parse(channel.attributes);
+    const result = await flexToCustomChannel.redirectMessageToExternalChat(context, {
+      event: sanitizedEvent,
+      recipientId,
+      sendExternalMessage: sendTwitterMessage(context),
+    });
 
-      // Redirect bot, system or third participant
-      if (channelAttributes.from !== event.From) {
-        const TwitResponse = await sendTwitterMessage(context, recipientId, Body);
-        console.log('Message sent from API call: ', Body);
-        resolve(success(TwitResponse));
+    switch (result.status) {
+      case 'sent':
+        resolve(success(result.response));
         return;
-      }
-
-      console.log('Message ignored (do not re-send self messages)');
-      resolve(success('Ignored event.'));
-      return;
+      case 'ignored':
+        resolve(success('Ignored event.'));
+        return;
+      default:
```

Comment:
Should the default clause be necessary? I thought the types `redirectMessageToExternalChat` was permitted to return would ensure it can't be anything other than the 2 cases?